atom/browser/lib/desktop-capturer.coffee

```diff
@@ -0,0 +1,38 @@
+ipc = require 'ipc'
+BrowserWindow = require 'browser-window'
+
+# The browser module manages all desktop-capturer moduels in renderer process.
+desktopCapturer = process.atomBinding('desktop_capturer').desktopCapturer
+
+getWebContentsFromId = (id) ->
+  windows = BrowserWindow.getAllWindows()
+  return window.webContents for window in windows when window.webContents?.getId() == id
+
+# The set for tracking id of webContents.
+webContentsIds = new Set
+
+stopDesktopCapture = (id) ->
+  webContentsIds.delete id
+  # Stop updating if no renderer process listens the desktop capturer.
+  if webContentsIds.size is 0
+    desktopCapturer.stopUpdating()
+
+# Handle `desktopCapturer.startUpdating` API.
+ipc.on 'ATOM_BROWSER_DESKTOP_CAPTURER_START_UPDATING', (event, args) ->
+  id = event.sender.getId()
+  if not webContentsIds.has id
+    # Stop sending desktop capturer events to the destroyed webContents.
+    event.sender.on 'destroyed', ()->
+      stopDesktopCapture id
+  # Start updating the desktopCapturer if it doesn't.
+  if webContentsIds.size is 0
+    desktopCapturer.startUpdating args
```

Comment:
A thing needs to be discussed here. In current implementation, the desktopCapturer in main process will ignore the `startUpdating` request when it has started, which will cause developer confusion in some scenarioes, considering:

1. A `dc.startUpdating({type:['screen', 'windows']})` is invoked in the first page of app.
2. A second `dc.startUpdating({type:['screen']})` is invoked in the second page of app.
3. Users find that the `windows` capturing result is also in second page of app. Because in main browser process, it ignores the step2's request.

And if we change to a workaround(stopping the current update, and then starting a new update), the result is also the same.